guides/v2.2/config-guide/secy/hashing.md

```diff
@@ -0,0 +1,57 @@
+---
+group: configuration-guide
+title: Password hashing
+---
+
+## Overview
+
+Currently Magento uses its own strategy for password hashing, based on different native PHP hashing algorithms. Magento supports multiple algorithms like `MD5`, `SHA256`, or`Argon 2ID13`. If the Sodium extension is installed (installed by default in PHP 7.2), then `Argon 2ID13` will be chosen as the default hashing algorithm. Otherwise, `SHA256` will be as default. As Magento still supports PHP 7.1.x, we cannot use the native PHP `password_hash` function with Argon 2i algorithm support, which was added in the PHP 7.2 release.
+
+To avoid compromising passwords which have been hashed with outdated algorithms like `MD5` and have not been updated in a while, the current implementation provides a method to upgrade the hash without changing the original password. In general, the password hash has the following format:
+
+```text
+password_hash:salt:version0:versionn
```

Comment:
Is the second `n` in `versionn` a variable? If yes, change it to `version<n>`

Reply:
Done.